pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -202,6 +202,8 @@ func BuildStorageFactory(masterConfig configapi.MasterConfig, server *kapiserver
 	// the order here is important, it defines which version will be used for storage
 	storageFactory.AddCohabitatingResources(batch.Resource("jobs"), extensions.Resource("jobs"))
 	storageFactory.AddCohabitatingResources(extensions.Resource("horizontalpodautoscalers"), autoscaling.Resource("horizontalpodautoscalers"))
+	// keep Deployments in extensions for backwards compatibility, we'll have to migrate at some point, eventually
```

Comment:
add to follow ups... we'll want to switch storage version to apps/v1beta1 in the same release as upstream so we're positioned to drop extensions/deployments when upstream does